### PR TITLE
Refactor source functional tests for static pages and session timeout

### PR DESCRIPTION
**Pull request by nabla-c0d3**
## Status

Ready

## Description of Changes

This PR continue the work from https://github.com/freedomofpress/securedrop/pull/6361, by bringing the new test code and fixtures to a few more functional tests for the source app.

More specifically, this PR:

* Refactors the functional test for when a source's session expires.
* Refactors a few test checking some of the source app's static pages.
* Moves the "pagelayout" tests to a sub-folder within the functional tests, so that fixtures can be shared (via conftest.py, to help with #3836). Fixtures were already shared previously but that was done by having a second class called `FunctionalTest`  inheriting from `functional_test.FunctionalTest`. Ultimately, most of the pagelayout tests are indeed functional tests and should be treated that way. This will also avoid having duplicate tests (one in functional, one in pagelayout, testing the exact same thing).

### CODE REVIEW

File: securedrop/tests/functional/pageslayout/test_journalist_account.py [hunk 1]
```diff
@@ -28,6 +29,9 @@ class TestJournalistLayoutAccount(
         source_navigation_steps.SourceNavigationStepsMixin,
         journalist_navigation_steps.JournalistNavigationStepsMixin):
 
+    # TODO(AD): This test fails often when run with the "ar" locale
+    #  The screenshot that gets saved is blank, which causes autocrop_btm() to fail
+    @flaky(max_runs=5)
```
Comment (nabla-c0d3) [inline securedrop/tests/functional/pageslayout/test_journalist_account.py:34]: This test is also flaky on the develop branch.

File: securedrop/tests/functional/pageslayout/test_journalist_account.py [hunk 2]
```diff
@@ -28,6 +29,9 @@ class TestJournalistLayoutAccount(
         source_navigation_steps.SourceNavigationStepsMixin,
         journalist_navigation_steps.JournalistNavigationStepsMixin):
 
+    # TODO(AD): This test fails often when run with the "ar" locale
+    #  The screenshot that gets saved is blank, which causes autocrop_btm() to fail
+    @flaky(max_runs=6)
```
Comment (nabla-c0d3) [inline securedrop/tests/functional/pageslayout/test_journalist_account.py]: This test is also flaky on the develop branch.

**Comment by nabla-c0d3** [comment]
Ready for review 👍

**Comment by zenmonkeykstop** [comment]
Tests pass happily but (at least for me) the `en_US` pagelayout screenshots have unicode markers in place of text characters. Will update with a proper review after poking about a bit.

**Comment by zenmonkeykstop** [comment]
Tests pass happily but (at least for me) the `en_US` pagelayout screenshots have unicode markers in place of text characters when run in the context of the full test suite. Will update with a proper review after poking about a bit.

**Comment by nabla-c0d3** [comment]
@zenmonkeykstop Sorry I should have checked the screenshots. Are you seing the same problem as in https://github.com/freedomofpress/securedrop/issues/4558 ?

**Comment by zenmonkeykstop** [comment]
Looked a bit more, it doesn't seem to be. But as it turn out, my apologies, this wasn't actually introduced by your changes, the same behavior is present in current `develop` and the last release tag as well. It's not a merge blocker - the only time the screenshots are relevant are when they're used to update the docs, and since the page layout tests are run separately for that the issue doesn't appear.

**Review by zenmonkeykstop — DISMISSED** [review]
Changes make sense and tests are passing. 

There is one unintended consequence in CI - the `app-page-layout-tests` job runs the entire test suite, as CircleCI's cli tool tries to find tests matching `/tests/pageslayout/test*py`, gets no result, and then that gets interpreted as a match against everything. I've pushed a commit to fix, if it works then this is good to go.

**Review by zenmonkeykstop — APPROVED** [review]
Changes make sense and tests are passing.

There is one unintended consequence in CI - the app-page-layout-tests job runs the entire test suite, as CircleCI's cli tool tries to find tests matching /tests/pageslayout/test*py, gets no result, and then that gets interpreted as a match against everything. I've pushed a commit to fix, if it works then this is good to go.

(Updating to add another fix to `make safety`, required to get CI passing)

**Comment by nabla-c0d3** [comment]
@zenmonkeykstop Thanks for reviewing this! Next PR is at https://github.com/freedomofpress/securedrop/pull/6379 👍

**Pull request merged**